File: leetcode/121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.java
// Last updated: 29/04/2026, 23:12:34
class Solution {
    public int maxProfit(int[] prices) {
        int min = Integer.MAX_VALUE;
        int max = 0;
        for(int i=0; i<prices.length; i++) {
            if(prices[i] < min) {
                min = prices[i];
            } else {
                max = Math.max(prices[i]-min, max);
            }
        }
        return max;
    }
}

File: leetcode/121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.cpp
// Last updated: 29/04/2026, 23:12:44
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if (prices.size() < 2) {
            return 0;
        }
        int maxProfit = 0;

        for (int i = 0, j = 1; i < prices.size() && j < prices.size(); j++) {
            int currentProfit = prices[j] - prices[i];
            maxProfit = max(maxProfit, currentProfit);

            if (prices[i] > prices[j]) {
                i = j;
            }
        }

        return maxProfit;
    }
};